virtual_ecosystem/core/config_builder.py

```diff
@@ -0,0 +1,584 @@
+"""The :mod:`~virtual_ecosystem.core.config_builder` provides tools to load a set of
+TOML formatted configuration dictionaries, either from files or from strings. String
+inputs are primarily intended for use in configuring models for testing, where it is
+more convenient to simply provide a string.
+
+The main class :class:`ConfigLoader` handles the loading of configuration data and
+compiling multiple sources into a single dictionary of configuration data. The public
+:meth:`ConfigLoader.get_configuration` method then passes the compiled data in the model
+instance through the :method:`generate_configuration` function to return the actual
+configuration object. Canonical use looks like this:
+
+.. code-block:: python
+
+    config_object = ConfigurationLoader(...).get_configuration()
```

Comment:
You can equally do:

```suggestion
    config_object = get_configuration(ConfigurationLoader(...).data)
```
And save an unnecessary method.

Indeed, this kind of usage of classes where the generated object is directly ditched just after its creation, using ever a function or attribute of the object, [is often considered an anti pattern](https://en.wikipedia.org/wiki/Poltergeist_(computer_programming)). Why creating an object that you are not using anywhere else?

I would suggest you re-write this using normal functions, passing arguments as needed.

Reply:
Definitely true on the `get_configuration` method - will make that change. I can definitely see what you mean about the `ConfigurationLoader` class being a poltergeist. I'm going to make that a new issue - I'd like to get the new system pushed through so would prefer to park that for now.